RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate Redux to Typescript PoC

## Proposed changes


Creation of a POC about migration from redux to typescript along with the tests.
And migrate SelectedUsersView, removing dispatch actions and setting new base types.

## Issue(s)	


## How to test or reproduce


go to:
New Message -> Create Team
Select/Deselect some users and click next
Create the team and see if users were sent via payload.


## Screenshots

## Types of changes




## Checklist



## Further comments
